src/qpmodel.jl

```diff
@@ -54,76 +66,108 @@ end
 
 linobj(qp::AbstractQuadraticModel, args...) = qp.data.c
 
-function objgrad(qp::AbstractQuadraticModel, x::AbstractVector)
+function objgrad(qp :: AbstractQuadraticModel, x :: AbstractVector)
     g = Vector{eltype(x)}(length(x))
     objgrad!(qp, x, g)
 end
 
-function objgrad!(qp::AbstractQuadraticModel, x::AbstractVector, g::AbstractVector)
+function objgrad!(qp :: AbstractQuadraticModel, x :: AbstractVector, g :: AbstractVector)
     v = qp.data.opH * x
     @. g = qp.data.c + v
     f = qp.data.c0 + dot(qp.data.c, x) + 0.5 * dot(v, x)
     qp.counters.neval_hprod += 1
     (f, g)
 end
 
-function obj(qp::AbstractQuadraticModel, x::AbstractVector)
+function obj(qp :: AbstractQuadraticModel, x :: AbstractVector)
     v = qp.data.opH * x
     f = qp.data.c0 + dot(qp.data.c, x) + 0.5 * dot(v, x)
     qp.counters.neval_hprod += 1
     f
 end
 
-function grad(qp::AbstractQuadraticModel, x::AbstractVector)
+function grad(qp :: AbstractQuadraticModel, x :: AbstractVector)
     g = Vector{eltype(x)}(undef, qp.meta.nvar)
     grad!(qp, x, g)
 end
 
-function grad!(qp::AbstractQuadraticModel, x::AbstractVector, g::AbstractVector)
+function grad!(qp :: AbstractQuadraticModel, x :: AbstractVector, g :: AbstractVector)
     v = qp.data.opH * x
     @. g = qp.data.c + v
     qp.counters.neval_hprod += 1
     g
 end
 
-hess_coord(qp::AbstractQuadraticModel, ::AbstractVector; kwargs...) = findnz(qp.data.H)
+hess_coord(qp :: AbstractQuadraticModel, ::AbstractVector; kwargs...) = findnz(qp.data.H)
 
-hess(qp::AbstractQuadraticModel, ::AbstractVector; kwargs...) = qp.data.H
+hess(qp :: AbstractQuadraticModel, ::AbstractVector; kwargs...) = qp.data.H
 
-hess_op(qp::AbstractQuadraticModel, ::AbstractVector; kwargs...) = qp.data.opH
+hess_op(qp :: AbstractQuadraticModel, ::AbstractVector; kwargs...) = qp.data.opH
 
-function cons(qp::AbstractQuadraticModel, x::AbstractVector)
+function cons(qp::AbstractQuadraticModel, x :: AbstractVector)
     c = Vector{eltype(x)}(undef, qp.meta.ncon)
     cons!(qp, x, c)
 end
 
-function cons!(qp::AbstractQuadraticModel, x::AbstractVector, c::AbstractVector)
+function cons!(qp :: AbstractQuadraticModel, x :: AbstractVector, c :: AbstractVector)
     mul!(c, qp.data.A, x)
     qp.counters.neval_jprod += 1
     c
 end
 
-jac_coord(qp::AbstractQuadraticModel, ::AbstractVector; kwargs...) = findnz(qp.data.A)
+"""
+Return the structure of the constraint's Jacobian in sparse coordinate format in place.
+"""
+function jac_structure!(qp :: QuadraticModel, rows :: Vector{<: Integer}, cols :: Vector{<: Integer}; kwargs...)
+    rows .= qp.data.A.rowval
+    cols .= findnz(qp.data.A)[2]
+end
+
+"""
+Return the structure of the Lagrangian Hessian in sparse coordinate format in place.
+"""
+function hess_structure!(qp :: QuadraticModel, rows :: Vector{<: Integer}, cols :: Vector{<: Integer}; kwargs...)
+    rows .= qp.data.H.rowval
+    cols .= findnz(qp.data.H)[2]
+end
```

Comment:
Move this `hess` function to right before `hess_coord!`.

Reply:
Also, neither this nor `jac_structure` has `kwargs`.

Reply:
> Move this `hess` function to right before `hess_coord!`.

Do you mean `hess(qp :: AbstractQuadraticModel, ::AbstractVector; kwargs...) = qp.data.H`? I changed it but don't understand why.

Reply:
I mean Hessian-related functions stay close together in the file, so move this one to right before where `hess_coord!` is defined.

Reply:
I'm not seeing the full file, but essentially is move Hessian functions close together and Jacobian functions close together.

Reply:
I was confused because hess and jac functions are all mixed up. Makes sense now. I'll put them in some logic order in a new commit.